examples/sat/bench_schoening.py

```diff
@@ -0,0 +1,187 @@
+import pandas as pd
+from bench_hillclimber import setup_default_logger
+from sat import SatInstance
+from bruteforce import bruteforce_solve
+from qubrabench.benchmark import track_queries, named_oracle
+from schoening import (
+    schoening_solve,
+    schoening_bruteforce_steps,
+)
+
+from qubrabench.utils.plotting import PlottingStrategy
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+@click.option(
+    "-n",
+    help="Number of variables.",
+    type=int,
+    required=True,
+)
+@click.option(
+    "-r",
+    help="Number of clauses divided by number of variables.",
+    type=int,
+    required=True,
+)
+@click.option("-strategy", type=str, default=None, show_default=True)
+@click.option(
+    "--seed",
+    help="Seed for the random operations.",
+    type=int,
+    required=False,
+)
+@click.option(
+    "--runs",
+    help="Number of runs (repetitions).",
+    default=10,
+    show_default=True,
+)
+@click.option(
+    "--verbose/--no-verbose",
+    help="Show more output.",
+    default=True,
+)
+@click.option(
+    "--save",
+    "dest",
+    help="Save to JSON file (preserves existing data!).",
+    type=click.Path(dir_okay=False, writable=True, path_type=Path),
+)
+def generate(r, seed, n, runs, dest, verbose, strategy):
+    """
+    Benchmark Schoening's algorithm for both strategies, bruteforcing assignments and steps.
+
+    Example:
+
+        bench_schoening.py --strategy=steps -r 3 -n 100 --save results.json
+    """
+    setup_default_logger(verbose)
+    rng = np.random.default_rng(seed)
+
+    history = []
+    for run in range(runs):
+        logging.debug(f"r={r}, n={n}, strategy={strategy}, #{run}")
+
+        # as schoening uses 3-SAT, k=3
+        inst = SatInstance.random(k=3, n=n, m=r * n, rng=rng)
+        with track_queries() as tracker:
+            # TODO: streamline selection of strategies...
+            if strategy == "standard":
+                schoening_solve(inst, error=10**-5, rng=rng)
+            elif strategy == "steps":
+                schoening_bruteforce_steps(inst, error=10**-5, rng=rng)
+            elif strategy is None:
+                bruteforce_solve(
+                    inst,
+                    (named_oracle("inst.evaluate"))(lambda k: inst.evaluate(k)),
+                    error=10**-5,
+                    rng=rng,
+                )
+
+            stats = tracker.get_stats("inst.evaluate")
+
+            # save record to history
+            rec = asdict(stats)
+            rec["n"] = n
+            rec["r"] = r
+            rec["strategy"] = strategy
+            rec["classical_control_method_calls"] = tracker.get_stats(
+                "inst.evaluate"
+            ).classical_actual_queries
+            history.append(rec)
+
+    # return pandas dataframe
+    history = pd.DataFrame(
+        [list(row.values()) for row in history], columns=list(history[0].keys())
+    )
+
+    history.insert(0, "impl", "bruteforce sat" if strategy is None else strategy)
```

Comment:
use the same key name in both places? i.e. `strategy` currently. But I think `variant` is more apt.